src/components/dashboard/subsettings/VaultSettings.tsx

```diff
@@ -28,26 +38,35 @@ export default function VaultSettings({ initial = false, vaultName }: Props) {
   );
 
   const [walletItems, setWalletItems] = useState<React.ReactNode[]>([]);
+  const [vaultBalance, setVaultBalance] = useState<number>(0);
 
   useEffect(() => {
     const loadWallets = async () => {
       if (!wallets) return;
 
       const walletPromises = wallets.map(async (wallet) => {
-        const svgieAvatar = svgie(wallet.address, { asDataURI: true });
         const gateway = await findGateway({});
         const arweave = new Arweave(gateway);
         const balance =
           arweave.ar.winstonToAr(
             await arweave.wallets.getBalance(wallet.address)
           ) || 0;
+        const svgieAvatar = svgie(wallet.address, { asDataURI: true });
         return (
           <ListItem
-            // TODO: Add click handler here
             key={wallet.address}
             title={`${wallet.nickname} (${formatAddress(wallet.address, 3)})`}
             description={`${formatTokenBalance(balance)} AR`}
             active={false}
+            onClick={() => {
+              // Switch active accounts
+              setActiveAddress(wallet.address);
+              browser.tabs.create({
+                url: browser.runtime.getURL(
+                  `popup.html?expanded=true#/send/transfer/AR/${vaultAddress}`
```

Comment:
FYI: this is getting blocked in Arc Browser, complaining that the "popup" can only be opened in the actual popup and we should use one of the "tabs" routes to open extension urls in windows

Reply:
@7i7o good to know - we'll have to revisit this with the expanded view as well